warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoWaveFunctions.java

```diff
@@ -56,19 +66,52 @@ private static boolean intersectsGeometries(Iterable<?> values, Geometry[] geome
     private static Geometry getGeometryFromFieldValue(Object fieldValue) {
         if (fieldValue instanceof Geometry) {
             return (Geometry) fieldValue;
+        } else if (fieldValue instanceof GeoNormalizer.GeoPoint) {
+            return geoPointToGeometry((GeoNormalizer.GeoPoint) fieldValue);
         } else if (fieldValue instanceof String) {
-            return AbstractGeometryNormalizer.parseGeometry((String) fieldValue);
+            return parseGeometry((String) fieldValue);
         } else if (fieldValue instanceof ValueTuple) {
             ValueTuple t = (ValueTuple) fieldValue;
             Object o = t.second();
             if (o instanceof AbstractGeometryType) {
-                AbstractGeometryType gt = (AbstractGeometryType) o;
-                return ((AbstractGeometry) gt.getDelegate()).getJTSGeometry();
+                AbstractGeometryType<?> gt = (AbstractGeometryType<?>) o;
+                return ((AbstractGeometry<?>) gt.getDelegate()).getJTSGeometry();
+            } else if (o instanceof GeoType) {
+                return parseGeometryFromGeoType(ValueTuple.getNormalizedStringValue(fieldValue));
             }
+        } else if (fieldValue instanceof FunctionalSet) {
+            FunctionalSet<?> funcSet = (FunctionalSet<?>) fieldValue;
+            Geometry[] geometries = funcSet.stream().map(GeoWaveFunctions::getGeometryFromFieldValue).toArray(Geometry[]::new);
+            return new GeometryCollection(geometries, geometryFactory);
         }
         throw new IllegalArgumentException("Field Value:" + fieldValue + " cannot be recognized as a geometry");
     }
     
+    private static Geometry parseGeometry(String geomString) {
+        Geometry geom;
+        try {
+            geom = AbstractGeometryNormalizer.parseGeometry(geomString);
+        } catch (IllegalArgumentException e) {
+            geom = parseGeometryFromGeoType(geomString);
+        }
+        return geom;
+    }
+    
+    private static Geometry parseGeometryFromGeoType(String geoTypeString) {
+        Geometry geom = null;
+        try {
+            geom = geoPointToGeometry(GeoNormalizer.isNormalized(geoTypeString) ? GeoNormalizer.GeoPoint.decodeZRef(geoTypeString) : GeoNormalizer.GeoPoint
+                            .decodeZRef(geoNormalizer.normalize(geoTypeString)));
+        } catch (Exception e) {
+            // do nothing
```

Comment:
really?

Reply:
we swallow the exception during the initial AbstractGeometryNormalizer.parseGeometry call on line 93 so that we can attempt to parse as GeoType via this function.  If the result of both of those is a null geometry, we throw an exception.  Do you want me to log something?  There currently isn't a logger in this class (I'm assuming for evaluation performance reasons?)